### PR TITLE
UefiPayloadPkg: FDT Support Phase 0

### PR DESCRIPTION
**Pull request by LinusvPelt**
# Description

<_Include a description of the change and why this change was made._>

<_For each item, place an "x" in between `[` and `]` if true. Example: `[x]` (you can also check items in GitHub UI)_>

<_Create the PR as a Draft PR if it is only created to run CI checks._>

<_Delete lines in \<\> tags before creating the PR._>

- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

<_Describe the test(s) that were run to verify the changes._>

## Integration Instructions

<_Describe how these changes should be integrated. Use N/A if nothing is required._>

### CODE REVIEW

**Comment by lgao4** [comment]
The change in MdePkg is good to me.

File: UefiPayloadPkg/UefiPayloadPkg.dec [hunk 1]
```diff
@@ -75,3 +87,18 @@ gUefiPayloadPkgTokenSpaceGuid.PcdBootManagerEscape|FALSE|BOOLEAN|0x00000020
 ## FFS filename to find the default variable initial data file.
 # @Prompt FFS Name of variable initial data file
  gUefiPayloadPkgTokenSpaceGuid.PcdNvsDataFile |{ 0x1a, 0xf1, 0xb1, 0xae, 0x42, 0xcc, 0xcf, 0x4e, 0xac, 0x60, 0xdb, 0xab, 0xf6, 0xca, 0x69, 0xe6 }|VOID*|0x00000025
+
+
+## Indicates if Universal payload support FDT
+#-  PcdHandOffFdtEnable is TRUE, HandOffData is FDT
+#-  PcdHandOffFdtEnable is FALSE, HandOffData is HOB
+gUefiPayloadPkgTokenSpaceGuid.PcdHandOffFdtEnable|TRUE|BOOLEAN|0x00000026
```
Comment (ChaselChiu) [inline UefiPayloadPkg/UefiPayloadPkg.dec:95]: Let's start with PcdHandOffFdtEnable = FALSE as default to not break existing platforms. Once those platforms are ready to use FDT they will toggle this PCD to TRUE. Later we may switch back to TRUE as default.

Comment (ChaselChiu) [inline UefiPayloadPkg/UefiPayloadPkg.dec:95]: Let's start with PcdHandOffFdtEnable = FALSE as default to not break existing platforms. Once those platforms are ready to use FDT they will toggle this PCD to TRUE. Later we may switch back to TRUE as default.

**Review by ChaselChiu — CHANGES_REQUESTED** [review]

File: UefiPayloadPkg/Include/Guid/UniversalPayloadBase.h [hunk 2]
```diff
@@ -16,4 +16,14 @@ typedef struct {
   EFI_PHYSICAL_ADDRESS                Entry;
 } UNIVERSAL_PAYLOAD_BASE;
 
+#define UNIVERSAL_PAYLOAD_BASE_REVISION  1
+
+#define N_NON_RELOCATABLE         BIT31
+#define P_NON_PREFETCHABLE        BIT30
```
Comment (ChaselChiu) [inline UefiPayloadPkg/Include/Guid/UniversalPayloadBase.h:22]: BIT30 set means "prefetchable", please correct the MACRO naming.

**Review by ChaselChiu — CHANGES_REQUESTED** [review]

File: UefiPayloadPkg/UefiPayloadPkg.dec (hunk 1 above)
Comment (ChaselChiu) [reply UefiPayloadPkg/UefiPayloadPkg.dec]: resolved

File: UefiPayloadPkg/Include/Guid/UniversalPayloadBase.h (hunk 2 above)
Comment (ChaselChiu) [reply UefiPayloadPkg/Include/Guid/UniversalPayloadBase.h]: resolved

**Review by ChaselChiu — APPROVED** [review]

**Comment by ChaselChiu** [comment]
Close this PR as we do not need 2 phases merging anymore. A single PR which incorporated all the changes has been merged: https://github.com/tianocore/edk2/pull/6095

**Pull request closed without merging**